cmd/conduit/root/pipelines/list.go

```diff
@@ -0,0 +1,118 @@
+// Copyright © 2025 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pipelines
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/alexeyco/simpletable"
+	"github.com/conduitio/conduit/cmd/conduit/api"
+	"github.com/conduitio/conduit/cmd/conduit/root/run"
+	apiv1 "github.com/conduitio/conduit/proto/api/v1"
+	"github.com/conduitio/ecdysis"
+	healthgrpc "google.golang.org/grpc/health/grpc_health_v1"
+)
+
+var (
+	_ ecdysis.CommandWithExecute = (*ListCommand)(nil)
+	_ ecdysis.CommandWithAliases = (*ListCommand)(nil)
+	_ ecdysis.CommandWithDocs    = (*ListCommand)(nil)
+	// with API client ?
+)
+
+type ListCommand struct {
+	client *api.Client
+	RunCmd *run.RunCommand
+}
+
+func (c *ListCommand) Docs() ecdysis.Docs {
+	return ecdysis.Docs{
+		Short: "List existing Conduit pipelines",
+		Long: `This command requires Conduit to be already running since it will list all pipelines registered 
+by Conduit. This will depend on the configured pipelines directory, which by default is /pipelines; however, it could 
+be configured via --pipelines.path at the time of running Conduit.`,
+		Example: "conduit pipelines ls",
+	}
+}
+
+func (c *ListCommand) Aliases() []string { return []string{"ls"} }
+
+func (c *ListCommand) Usage() string { return "list" }
+
+func (c *ListCommand) Execute(ctx context.Context) error {
+	// TODO: Move this elsewhere since it'll be common for all commands that require having Conduit Running
+	// --------- START
+	conduitNotRunning := "Notice: To inspect the API, Conduit needs to be running" +
+		"\nPlease execute `conduit run`"
+
+	conduitGRPCAddr := c.RunCmd.GRPCAddress()
+
+	conduitClient, err := api.NewClient(ctx, conduitGRPCAddr)
```

Comment:
Not sure if possible, but we could think about injecting the client with a decorator through ecdysis. Similarly how we will inject a logger. Didn't we do something similar in the meroxa CLI?

Reply:
@lovromazgon yup. Been thinking about it. I think I'll do something similar https://github.com/meroxa/cli/blob/5cfa2489a37590377ff021a6c867724b31e4d149/cmd/meroxa/builder/builder.go#L67-L71. Had [a reminder](https://github.com/ConduitIO/conduit/pull/2063#discussion_r1911067810) for now, but I'll make it part of this.